### ISSUE TITLE
roving-tabindex-controller ignores for expandable items

### ISSUE THREAD

**Issue by nikkimk** [patternfly/patternfly-elements#2408]
## roving-tabindex-controller ignores for expandable items

For components with complex navigation, such as navigation with expandable submenus, roving-tabindex controller has the following issues:
- listeners are set to host when only the items should have listeners.
- vertical navigation is disabled for `aria-expanded=true` items when navigation should be enabled and include navigation to any children that are on its items list.
- currently allows navigation for `aria-expanded=false` items when it should skip children

### Steps to reproduce

1. Go to the [rh-menu demo](https://deploy-preview-679--red-hat-design-system.netlify.app/components/menu/demo/).
2. Using a keyboard focus on the menu button while closed.
3. Press the down arrow. Nothing should happen.
4. Press enter to open the menu.
5. Press the down arrow. Focus should be set on the active subitem.

### Expected behavior

1. Go to the [rh-menu demo](https://deploy-preview-679--red-hat-design-system.netlify.app/components/menu/demo/).
2. Using a keyboard focus on the menu button while closed.
3. Press the down arrow. 
4. If item is part of a larger roving tabindex group, like a parent menu, focus should be on the next sibling. If item is *not* part of a larger roving tabindex group, nothing should happen. 
5. Press enter to open the menu.
6. Press the down arrow. Focus should be set on the active subitem.
